File: python/1342-number-of-steps-to-reduce-a-number-to-zero.py
class Solution:
    def numberOfSteps (self, num):
      steps = 0
      while num > 0:
        steps += 1
        if num % 2 != 0:
          num -= 1
          continue

        if num == 0:
          break
        
        num = int(num / 2)
      
      return steps

print(Solution().numberOfSteps(14))

File: javascript/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
  let steps = 0;
  while (num > 0) {
    steps++;
    if (num % 2 !== 0) {
      num--;
      continue;
    }

    if (num === 0) {
      break;
    }

    num /= 2;
  }

  return steps;
};
